vault/resource_aws_secret_backend_role.go

```diff
@@ -97,19 +116,47 @@ func awsSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
 		policy = d.Get("policy")
 	}
 
-	if policy == "" && len(policyARNs) == 0 {
-		return fmt.Errorf("either policy or policy_arn must be set.")
+	var roleARNs []string
+	credentialType := d.Get("credential_type").(string)
+
+	if credentialType == "assumed_role" {
+		roleARNsIfc, ok := d.GetOk("role_arns")
+		if ok {
+			for _, arnIfc := range roleARNsIfc.([]interface{}) {
+				roleARNs = append(roleARNs, arnIfc.(string))
+			}
+		}
+		if len(roleARNs) == 0 {
+			return fmt.Errorf("role_arns must be set if credential_type is assumed_role")
+		}
+	}
+
+	if policy == "" && len(policyARNs) == 0 && len(roleARNs) == 0 {
+		return fmt.Errorf("either policy or policy_arn or role_arns must be set")
```

Comment:
The error is misleading, you are checking policy, policyARN and roleARN but mentioning only 2 of them in the error

Reply:
thanks for the review, will address shortly